3-make_gene_matrix.R

```diff
@@ -0,0 +1,122 @@
+  source("https://bioconductor.org/biocLite.R")
+  biocLite("org.Hs.eg.db", suppressUpdates = TRUE)
+}
+if (!("tximport" %in% installed.packages())) {
+  source("https://bioconductor.org/biocLite.R")
+  biocLite("tximport", suppressUpdates = FALSE)
+}
+if (!("rjson" %in% installed.packages())) {
+  install.packages("rjson", suppressUpdates = FALSE)
+}
+
+# Attach needed libraries
+library(org.Hs.eg.db)
+library(optparse)
+library(tximport)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+#--------------------------------Set up options--------------------------------#
+option_list <- list(
+    make_option(opt_str = c("-d", "--dir"), type = "character", default = getwd(),
+        help = "Directory where salmon quantification folders are located",
+        metavar = "character"),
+    make_option(opt_str = c("-g", "--geo"), type = "character", default = getwd(),
+        help = "GEO ID of dataset for downloading the metadata",
+        metavar = "character"),
+    make_option(opt_str = c("-o", "--output"), type = "character",
+        default = getwd(), help = "Directory where you would like the output to go",
+        metavar = "character"),
+    make_option(opt_str = c("-m", "--mapped"), type = "numeric", 
+        default = "0.5", help = "Cutoff for what percent mapped_reads samples
+        should have. Any samples with less than the cutoff will be removed.",
+        metavar = "numeric"), 
+    make_option(opt_str = c("-l", "--label"), type = "character",
+        default = "", help = "Optional label for output files",
+        metavar = "character")
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+#------------------------------Import Salmon reads-----------------------------#
+# Get quant files
+quant.files <- list.files(opt$dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+
+# Carry sample names over
+sample.names <- stringr::word(quant.files, 3, sep = "/")
+
+# Get transcript IDs
+transcripts <- read.table(quant.files[[1]], header = TRUE, 
+                 colClasses = c("character", rep("NULL", 4)))
+
+transcripts.wout.ver <- gsub("\\.[0-9]*$", "", transcripts$Name)
+
+# Turn transcript ids to gene ids
+tx.gene.key <- data.frame('transcript' = transcripts$Name,
+                          'gene' = mapIds(org.Hs.eg.db, keys = transcripts.wout.ver,
+                                         column = "ENSEMBL", keytype = "ENSEMBLTRANS"),
+                          stringsAsFactors = FALSE)
+
+# Remove transcripts without genes
+tx.gene.key <- tx.gene.key[!is.na(tx.gene.key$gene),]
+
+# Do the thing
+tx.counts <- tximport::tximport(quant.files, type = "salmon", tx2gene = tx.gene.key,
+                                countsFromAbundance = "no")
+
+# make as a dataframe
+tx.counts <- data.frame(tx.counts$counts, stringsAsFactors = FALSE)
+
+# Get gene symbols for data
+gene <- mapIds(org.Hs.eg.db, keys = tx.gene.key$gene , column = "SYMBOL",
```

Comment:
Does this get used anywhere else?

Reply:
Not yet, but I might be using it in the future. Not sure yet.

Reply:
I will delete for now.